### DISCUSSION TITLE
ESP8266EX chip flashed with micropython issue: trying to run boot.py and main.py with simple built in led flash but not working

### DISCUSSION THREAD

**Discussion by Alejo-Correa** [micropython/micropython#11677] [ESP8266]
Hi everyone,

I am new to this world of microcontrollers so I just want to build knowledge from practical experience and discussions. I have an ESP8266EX board that includes a 0.96" OLED display. This board had originally a different flash file that was supposed to be for using with an Arduino IDE. I proceeded to erase and re install the micropython flash .bin via esptool that I set using a Windows cmd. Well just to summarise, I managed to get all the directories sorted and packages installed, I'm using PuTTy as a terminal emulator and ampy as the "file manager" tool. Initially  I looked for a simple code to make the built in led to intermittently flash but when accessing via PuTTY, I run the code, the main.py,  and the terminal goes crazy returning endless continuous rubbish while the led on the board remains lit, which is obviously not the right response. The issue I got is that when I uploaded the boot.py files and main.py files to the board these don't seem to be doing anything when soft resetting or re connecting to the port. Find below the scripts I have loaded into the board and maybe a more experienced person can tell me what I might be doing wrong:

boot.py:
```py
import esp
esp.osdebug(None)
import uos, machine
#uos.dupterm(None, 1) # disable REPL on UART(0)
import gc
#import webrepl
#webrepl.start()
gc.collect()
```

main.py
```py
import machine
import time

pin = machine.Pin(0, machine.Pin.OUT)
for i in range(10):
    pin.on()
    time.sleep(0.5)
    pin.off()
    time.sleep(0.5)
```

Cheers,
  
_(Edited by jimmo to add code formatting)_

**Comment by jimmo** [comment]
hi @Alejo-Correa , please see [Welcome (Please read before posting)](https://github.com/orgs/micropython/discussions/9111) for instructions on how to format code in posts.

My guess would be that PuTTY is either using the wrong serial port or the wrong baud rate.

Instead of using `ampy` and PuTTY, we recommend using [mpremote](https://docs.micropython.org/en/latest/reference/mpremote.html) which includes the functionality of both and is specifically designed for MicroPython. The documentation is in the middle of being updated, see https://github.com/micropython/micropython/pull/11659 for the in-progress update.

**Reply by Alejo-Correa** [reply]
First of all, Thanks for the advice on this library, I will check it out. Second, Well I've read that micropython works with a baud rate of 115200 so that is the one I always input (I tried different values but I don't get a response in the cmd application nor in the terminal emulators) and the port, well that is not an option as the device manager tells me which COMX is and I have always used that value to get this far. Any other suggestions maybe?

**Comment by robert-hh** [comment]
If there is an endless trail of rubbish going on when connected to Putty, then most likely the firmware is not installed properly. Try uploading the firmware with different flash mode options of esptool.py, and specify the flash size and chip type.  The flash modes you can try are DIO and DOUT. As far as I recall, the default is DIO.
When the upload is fine, putty should show a short sequence of gibberish, and then stop with the REPL prompt.

**Comment by Alejo-Correa** [comment]
Hi, I just reinstalled the firmware and tried the `mperemote` package but still no luck with that particular task so I gave up on that project. Instead , I tried to activate the built in OLED screen on which I was successful. During this process I also realised that my board has actually a different arrangements of the pins, on the plan which is included in the insert of the board, so that is maybe why I failed previously. It is not stated the Pin 0 though. Thanks.

**Reply by silicon-systems** [reply]
Which firmware version are you using. Flash modes set QIO. I just flashed micropython ver. **esp8266-20230426-v1.20.0**
from the IDE.

Cheers Hanspeter.

**Reply by Alejo-Correa** [reply]
Yeah, that's exactly the one I'm using now.

**Comment by tobiasfs** [comment]
Hi,
I had a similar problem while working with some old hardware (ThunderTiger's RoboHero)
### Problem
ESP-WROOM-02 (with ESP8266EX inside)
Flashed firmware for ESP-WROOM-32 according to instructions
prompt displayed only gibberish with settings 115200-8-N-1.
### Solution
The ESP-WROOM-**32** is ESP32. These chips need a flash offset of **0x1000**.
The ESP-WROOM-**02** is ESP8266EX. These chips need a flash offset of **0**.
Flashing with 
`esptool --port /dev/ttyUSB0 --baud 46800 write_flash --flash_size=detect 0 ~/Downloads/ESP8266_GENERIC-20251209-v1.27.0.bin`
made the firmware work correctly and the prompt is correctly displayed with 115200 baud.
### Result
```
MicroPython v1.27.0 on 2025-12-09; ESP module with ESP8266
Type "help()" for more information.
>>>
```

**Reply by scruss** [reply]
> The ESP-WROOM-32 is ESP32. These chips need a flash offset of 0x1000.

And a completely different firmware for their completely different architecture.
